File: 0344-reverse-string/0344-reverse-string.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        int a = 0 ; 
        int n = s.size() ; 
        int b = n-1 ; 
        while ( a < b ){
        swap(s[a] , s[b]); 
        a++ ; 
        b-- ;             
        }
    }
};

File: 0344-reverse-string/0344-reverse-string.rs
impl Solution {
    pub fn reverse_string(s: &mut Vec<char>) {
        let mut a = 0;
        let mut b = s.len() - 1;
        
        while a < b {
            s.swap(a, b);
            a += 1;
            b -= 1;
        }
    }
}